app/settings/surveys/attribute-editor.directive.js

```diff
@@ -76,6 +76,63 @@ function (
             $scope.canDisableCaption = function () {
                 return $scope.editAttribute.type === 'media' && $scope.editAttribute.input === 'upload';
             };
+
+            $scope.canDisplayDefaultValue = function () {
+                return $scope.editAttribute.input !== 'upload' && $scope.editAttribute.type !== 'title' && $scope.editAttribute.type !== 'description' && $scope.editAttribute.input !== 'tags' && $scope.editAttribute.input !== 'location' && $scope.editAttribute.input !== 'relation';
+            };
+
+            $scope.errorMsgs = {
+                'num': 'Number(s) only',
+                'video': 'Enter valid video link',
+                'others': 'hidden'
+            };
+
+            $scope.inputType = [
+                $scope.editAttribute.input === 'number',
+                $scope.editAttribute.input === 'video'
+            ]
+
+            $scope.errorMessage = function () {
+
+                if ($scope.inputType[0]) {
+                    $scope.errorMsg = $scope.errorMsgs.num;
+                }
+
+                if ($scope.inputType[1]) {
+                    $scope.errorMsg = $scope.errorMsgs.video;
+                }
+
+                if (!$scope.errorMsg) {
+                    $scope.errorMsg = $scope.errorMsgs.others;
+                }
+
+                return $scope.errorMsg;
+            };
+
+            $scope.canDisplayError = function () {
+
+                if ($scope.inputType[0] && isNaN($scope.editAttribute.default) && $scope.editAttribute.default.length >= 1) {
+                   return angular.element(document.querySelector('#displayError')).removeClass('hidden') && angular.element(document.querySelector('#form-field')).addClass('error');
+                }
+
+                if ($scope.inputType[1] && $scope.editAttribute.default.length >= 1) {
```

Comment:
For validation of video-defaults, you could actually make use of the video-validation we have for editing posts: https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/modify/post-edit.service.js#L24, you can see how it is used here: https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/modify/post-video.directive.js#L65.

Reply:
> For validation of video-defaults, you could actually make use of the video-validation we have for editing posts: https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/modify/post-edit.service.js#L24, you can see how it is used here: https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/modify/post-video.directive.js#L65.

This is a great guide @Angamanga Thank you. Will work with this.